### PR TITLE
Updated Hero Sample to support PSTN Dial Out

### PR DESCRIPTION
**Pull request by mikehang-msft**
Updated calling components to the latest version

### CODE REVIEW

File: Rooms Hero Sample/ClientApp/src/Utils/Utils.ts [hunk 1]
```diff
@@ -55,15 +56,14 @@ export const utils = {
         }
         throw new Error('Invalid token response');
     },
-    createRoom: async (identity: string): Promise<string> => {
+    createRoom: async (identity: string, pstnDialOutEnabled: boolean): Promise<string> => {
         const requestOptions = {
             method: 'POST',
             headers: { 'Content-Type': 'application/json' },
             body: JSON.stringify({
```
Comment (allchiang-msft) [inline Rooms Hero Sample/ClientApp/src/Utils/Utils.ts]: We can simplify this line if the body is empty (was the empty body intentional?)

File: Rooms Hero Sample/Rooms Hero Sample.csproj [hunk 2]
```diff
@@ -12,7 +12,7 @@
   <ItemGroup>
     <PackageReference Include="Azure.Communication.Identity" Version="1.2.0" />
     <PackageReference Include="Azure.Communication.Common" Version="1.2.1" />
-    <PackageReference Include="Azure.Communication.Rooms" Version="1.0.0" />
+	<PackageReference Include="Azure.Communication.Rooms" Version="1.1.0-beta.1" />
```
Comment (allchiang-msft) [inline Rooms Hero Sample/Rooms Hero Sample.csproj]: nit: indent spacing

File: Rooms Hero Sample/Controllers/RoomServiceClientController.cs [hunk 3]
```diff
@@ -140,8 +141,15 @@ public async Task<IActionResult> CreateRoom(string userId)
                 CommunicationUserIdentifier identifier = new CommunicationUserIdentifier(userId);
                 RoomParticipant participant1 = new RoomParticipant(identifier);
                 List<RoomParticipant> invitedParticipants = new List<RoomParticipant> { participant1 };
+                CreateRoomOptions createOption = new CreateRoomOptions()
+                {
+                    ValidFrom = validFrom,
+                    ValidUntil = validUntil,
+                    Participants = invitedParticipants,
+                    PstnDialOutEnabled = pstnDialOutEnabled,
+                };
```
Comment (allchiang-msft) [inline Rooms Hero Sample/Controllers/RoomServiceClientController.cs:150]: The object of options should be inside the constructor params right?

File: Rooms Hero Sample/ClientApp/src/Utils/Utils.ts [hunk 4]
```diff
@@ -28,6 +28,8 @@ export type UpdateParticipant = {
     add: boolean;
 }
 
+export type Role = "Presenter" | "Attendee" | "Consumer"
```
Comment (allchiang-msft) [inline Rooms Hero Sample/ClientApp/src/Utils/Utils.ts:31]: why aren't we importing the roles anymore?

File: Rooms Hero Sample/ClientApp/src/Utils/Utils.ts (hunk 1 above)
Comment (mikehang-msft) [reply Rooms Hero Sample/ClientApp/src/Utils/Utils.ts]: there is no payload in this request besides the title which is not being used. Let me remove the body string all together.

File: Rooms Hero Sample/Controllers/RoomServiceClientController.cs (hunk 3 above)
Comment (mikehang-msft) [reply Rooms Hero Sample/Controllers/RoomServiceClientController.cs:150]: in this case is not, this is an internal web API wrapper to call into Rooms SDK. For the purpose of Hero sample, we are using default parameter values (validFrom, validUntil, participant, etc.). Hence we do not need to expose them.

File: Rooms Hero Sample/ClientApp/src/Utils/Utils.ts (hunk 1 above)
Comment (mikehang-msft) [reply Rooms Hero Sample/ClientApp/src/Utils/Utils.ts]: updated

File: Rooms Hero Sample/ClientApp/src/Utils/Utils.ts (hunk 4 above)
Comment (mikehang-msft) [reply Rooms Hero Sample/ClientApp/src/Utils/Utils.ts:31]: Role is no longer available in @Azure/communication-react package anymore. I think it makes sense to not reference @Azure/communication-react as those roles are more explicitly to Rooms

**Pull request merged**